api/src/test/java/org/apache/iceberg/metrics/TestScanReport.java

```diff
@@ -66,11 +83,12 @@ public void fromEmptyScanMetrics() {
     Schema projection =
         new Schema(Types.NestedField.required(1, "c1", Types.StringType.get(), "c1"));
     ScanReport scanReport =
-        ScanReport.builder()
-            .withTableName(tableName)
-            .withProjection(projection)
-            .withFilter(Expressions.alwaysTrue())
-            .fromScanMetrics(ScanReport.ScanMetrics.NOOP)
+        ImmutableScanReport.builder()
+            .tableName(tableName)
+            .snapshotId(-1L)
```

Comment:
Since this is not required (and probably should have been before) can we just set it to a reasonable value rather than passing through -1?

Reply:
done

Reply:
we could also add validation to make sure snapshotId is always > 0. All that would be required is adding the below method to `ScanReport`:

```
  @Check
  default void check() {
    Preconditions.checkArgument(snapshotId() > 0, "...");
  }
```